### PR TITLE
Npm: Handle file reference in declared license

### PR DESCRIPTION
**Pull request by fviernau**
Proposed as replacement for #2248.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Npm.kt [hunk 1]
```diff
@@ -168,9 +168,15 @@ open class Npm(
...
-            // NPM does not mean https://unlicense.org/ here, but the wish to not "grant others the right to use a
-            // private or unpublished package under any terms", which corresponds to SPDX's "NONE".
-            if (declaredLicense == "UNLICENSED") SpdxLicense.NONE else declaredLicense
+            when {
+                // NPM does not mean https://unlicense.org/ here, but the wish to not "grant others the right to use a
+                // private or unpublished package under any terms", which corresponds to SPDX's "NONE".
+                declaredLicense == "UNLICENSED" -> SpdxLicense.NONE
+                // NPM allows to declare non-SPDX licenses only by referencing a license file. Avoid reporting an
+                // [OrtIssue] by mapping this to a valid license identifier.
+                declaredLicense.startsWith("SEE LICENSE IN ") -> "LicenseRef-scancode-unknown-license-reference"
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Npm.kt]: IMO we should stay scanner-agnostic here and not use a LicenseRef from the `scancode` namespace here, but from our own `ort` namespace, i.e. `LicenseRef-ort-unknown-license-reference` or similar.

Comment (fviernau) [reply analyzer/src/main/kotlin/managers/Npm.kt]: I totally agree. However, looking at `DeclaredLicenseMapping` there are multiple mappings to `ScanCode` licenses and thus this change is consistent with that. I propose we create a ticket for removing ScanCode license identifiers from the analyzer consistently. Would you be fine with that approach?

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Npm.kt]: Yes, there are some such mappings in `SpdxDeclaredLicenseMapping` of the `spdx-utils` module that use the `licenseRef()` helper function. I agree these need to be fixed separately (and ASAP), but I don't see why that should justify to now introduce code to the `analyzer` module which we already know to be wrong. So I'd prefer to do this change right from the start.

I would have agreed with you if you had introduced another mapping to `SpdxDeclaredLicenseMapping` in `spdx-utils`, but this is something different to me.

Comment (fviernau) [reply analyzer/src/main/kotlin/managers/Npm.kt]: Done, even though I'm not really convinced.

File: analyzer/src/main/kotlin/managers/Npm.kt [hunk 2]
```diff
@@ -167,10 +167,16 @@ open class Npm(
...
-            // NPM does not mean https://unlicense.org/ here, but the wish to not "grant others the right to use a
-            // private or unpublished package under any terms", which corresponds to SPDX's "NONE".
-            if (it == "UNLICENSED") SpdxLicense.NONE else it
+        return declaredLicenses.mapTo(sortedSetOf()) { declaredLicense ->
+            when {
+                // NPM does not mean https://unlicense.org/ here, but the wish to not "grant others the right to use a
+                // private or unpublished package under any terms", which corresponds to SPDX's "NONE".
+                declaredLicense == "UNLICENSED" -> SpdxLicense.NONE
+                // NPM allows to declare non-SPDX licenses only by referencing a license file. Avoid reporting an
+                // [OrtIssue] by mapping this to a valid license identifier.
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Npm.kt]: Isn't the fact whether an OrtIssue is reported or not subject to evaluator rules, and thus user-specific?

Comment (fviernau) [reply analyzer/src/main/kotlin/managers/Npm.kt]: The `OrtIssue` I'm talking about has nothing to do with the `evaluator`.
It is in fact an `analyzer` issue.

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Npm.kt]: Oh, wow, I wasn't aware that we're flagging this even as an `ERROR`... thanks for reminding me of that code.

Comment (fviernau) [reply analyzer/src/main/kotlin/managers/Npm.kt]: Sure! Thanks for reviewing!

**Pull request merged**